app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -334,7 +334,20 @@ def template_from_vmdb(deployment)
 
       def download_template(uri)
         require 'open-uri'
-        open(uri) { |f| f.read }
+
+        if @config.proxy.blank?
+          open(uri) { |f| f.read }
+        else
+          proxy = Addressable::URI.parse(@config.proxy)
+          proxy_url = "#{proxy.scheme}://#{proxy.host}:#{proxy.port}"
```

Comment:
I blame open-uri for this, because it requires separate options if it's an authenticating proxy or not. Since the information is coming from an internal source (vmdb config), I think it's safe.